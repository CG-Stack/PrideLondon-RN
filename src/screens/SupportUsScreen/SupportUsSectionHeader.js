// @flow
import React from "react";
import { ImageBackground, StyleSheet } from "react-native";
import type { ImageRef } from "../../data/image-ref";
import Text from "../../components/Text";
import { lightNavyBlueColor, whiteColor } from "../../constants/colors";

type Props = {
  image: ImageRef,
  title: string
};

const SupportUsSectionHeader = ({ image, title }: Props) => (
  <ImageBackground style={styles.image} source={image} resizeMode="cover">
    <Text type="h1" style={styles.title}>
      {title}
    </Text>
  </ImageBackground>
);

const styles = StyleSheet.create({
  image: {
    aspectRatio: 2 / 1,
    justifyContent: "flex-end",
    alignItems: "flex-start"
  },
  title: {
    backgroundColor: whiteColor,
    color: lightNavyBlueColor,
    margin: 16,
    paddingHorizontal: 8,
    lineHeight: 40
  }
});

export default SupportUsSectionHeader;
