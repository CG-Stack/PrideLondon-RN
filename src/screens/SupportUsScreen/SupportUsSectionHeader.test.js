// @flow
import React from "react";
import { shallow } from "enzyme";
import SupportUsSectionHeader from "./SupportUsSectionHeader";

it("renders correctly", () => {
  const output = shallow(<SupportUsSectionHeader image={0} title="Fruit" />);
  expect(output).toMatchSnapshot();
});
