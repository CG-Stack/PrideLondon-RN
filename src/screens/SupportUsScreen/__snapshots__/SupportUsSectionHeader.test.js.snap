// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ImageBackground
  resizeMode="cover"
  source={0}
  style={
    Object {
      "alignItems": "flex-start",
      "aspectRatio": 2,
      "justifyContent": "flex-end",
    }
  }
>
  <Text
    markdown={false}
    style={
      Object {
        "backgroundColor": "#ffffff",
        "color": "#2d2f7f",
        "lineHeight": 40,
        "margin": 16,
        "paddingHorizontal": 8,
      }
    }
    type="h1"
  >
    Fruit
  </Text>
</ImageBackground>
`;
